include/private/rdycoreimpl.h

```diff
@@ -54,6 +54,9 @@ typedef struct {
   RDySedimentCondition *sediment;
   RDySalinityCondition *salinity;
 
+  // true if either RDyFlowCondition, RDySedimentCondition, or RDySalinityCondition is set
+  PetscBool is_defined;
```

Comment:
Is this preferable to checking the `flow`, `sediment`, and/or `saliinity` pointers for non-`NULL` values?